docs/content/Examples-Tutorials-Recipes/Snapshots.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Snapshots
+permalink: /recipes/snapshots
+category: Examples & Tutorials
+subCategory: Data schema
+menuOrder: 3
+---
+
+## Use case
+
+For a dataset that contains a sequence of changes to a property over time,
+we want to be able to get the most recent state of said property at any given date.
+In the recipe below we'll learn, for a cube with `Product Id`, `Status`, and
+`Changed At` dimensions, how to calculate the snapshots of statuses at any given date.
+
+<!-- prettier-ignore-start -->
+[[info | ]]
+| We can consider the status property to be a
+| [slowly changing dimension](https://en.wikipedia.org/wiki/Slowly_changing_dimension)
+| (SCD) of type 2. Modeling data schemas with slowly changing dimensions is an
+| essential part of the data engineering skillset.
+<!-- prettier-ignore-end -->
+
+## Data schema
+
+Let's explore the `Statuses` cube that contains data like this:
+
+```javascript
+[
+  {
+    "Statuses.orderId": 1,
+    "Statuses.status": "shipped",
+    "Statuses.changedAt": "2019-01-19T00:00:00.000"
+  },
+  {
+    "Statuses.orderId": 1,
+    "Statuses.status": "processing",
+    "Statuses.changedAt": "2019-03-14T00:00:00.000"
+  },
+  {
+    "Statuses.orderId": 1,
+    "Statuses.status": "completed",
+    "Statuses.changedAt": "2019-01-25T00:00:00.000"
+  },
+  {
+    "Statuses.orderId": 2,
+    "Statuses.status": "processing",
+    "Statuses.changedAt": "2019-08-21T00:00:00.000"
+  },
+  {
+    "Statuses.orderId": 2,
+    "Statuses.status": "completed",
+    "Statuses.changedAt": "2019-04-13T00:00:00.000"
+  },
+  {
+    "Statuses.orderId": 2,
+    "Statuses.status": "shipped",
+    "Statuses.changedAt": "2019-03-18T00:00:00.000"
+  }
+]
+```
+
+We can see that statuses change occasionally. How do we count orders that remained
+in the `shipped` status at a particular date?
+
+First, we need to generate a range with all dates of interest, from the earliest to
+the latest. Second, we need to join the dates with the statuses and leave only the
+most recent statuses to date. 
+
+```javascript
+cube(`StatusSnapshots`, {
+  extends: Statuses,
+
+  sql: `
+    -- Create a range from the earlist date to the latest date
+    WITH range AS (
+      SELECT date
+      FROM GENERATE_SERIES(
```

Comment:
this sql is unique to bigquery (and maybe supported by other databases), we need to mention it.

Reply:
It's Postgres-specific, but I get your point.